internal/cli/base.go

```diff
@@ -608,6 +619,47 @@ func checkFlagsAfterArgs(args []string, set *flag.Sets) error {
 	return nil
 }
 
+// workspace computes the workspace based on available values, in this order of
+// precedence (last value wins):
+//
+// - value stored in the CLI context
+// - vaule from the environment variable WP_WORKSPACE
+// - value set in the CLI flag -workspace
+//
+// The default value is "default"
+func (c *baseCommand) workspace() (string, error) {
+	// load env for workspace
+	workspaceENV := os.Getenv(defaultWorkspaceEnvName)
+	switch {
+	case c.flagWorkspace != "":
+		return c.flagWorkspace, nil
+	case workspaceENV != "":
+		return workspaceENV, nil
+	default:
+		// attempt to load from CLI context storage
+		defaultName, err := c.contextStorage.Default()
+		if err != nil {
+			return "", err
+		}
+
+		// If we have no context name, then we just return the default
+		if defaultName != "" && defaultName != "-" {
+			// Load the context and return the workspace value. If it's empty,
+			// we'll fall through and return the default
+			cfg, err := c.contextStorage.Load(defaultName)
+			if err != nil {
+				return "", err
+			}
+			if cfg.Workspace != "" {
+				return cfg.Workspace, nil
+			}
+		}
+	}
+
+	// default value
+	return defaultWorkspace, nil
```

Comment:
I think this is a bit easier to read if the default return value is inside of the `default` case.

Reply:
👍 update in 7886733d